src/safeds/ml/nn/_model.py

```diff
@@ -57,33 +71,39 @@ def fit(
             raise OutOfBoundsError(actual=epoch_size, name="epoch_size", lower_bound=ClosedBound(1))
         if batch_size < 1:
             raise OutOfBoundsError(actual=batch_size, name="batch_size", lower_bound=ClosedBound(1))
+        if train_data.features.number_of_columns is not self._input_size:
+            raise InputSizeError(train_data.features.number_of_columns, self._input_size)
+        self._feature_names = train_data.features.column_names
+        self._batch_size = batch_size
         copied_model = copy.deepcopy(self)
-        copied_model._batch_size = batch_size
-        dataloader = train_data._into_dataloader(copied_model._batch_size)
+
+        dataloader = train_data._into_dataloader_with_classes(copied_model._batch_size, 1)
 
         loss_fn = nn.MSELoss()
 
-        optimizer = torch.optim.SGD(copied_model._model.parameters(), lr=0.05)
-        loss_sum = 0.0
-        number_of_batches_done = 0
-        for epoch in range(epoch_size):
-            for x, y in dataloader:
+        optimizer = torch.optim.SGD(copied_model._model.parameters(), lr=learning_rate)
+        for _ in range(epoch_size):
+            for x, y in iter(dataloader):
                 optimizer.zero_grad()
 
                 pred = copied_model._model(x)
 
                 loss = loss_fn(pred, y)
-                loss_sum += loss.item()
+                self._loss_sum += loss.item()
```

Comment:
I think it would make more sense to reset the loss after each epoch. If you calculate the loss over the whole period of training, you cannot really compare two epochs on their loss.